clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleHttpLoadBalancerCachingAgent.groovy

```diff
@@ -0,0 +1,379 @@
+            portRange: forwardingRule.portRange,
+            healths: [],
+            hostRules: [],
+        )
+        loadBalancers << newLoadBalancer
+
+        if (forwardingRule.target) {
+          def targetProxyName = Utils.getLocalName(forwardingRule.target)
+          def targetProxyCallback = new TargetProxyCallback(
+              googleLoadBalancer: newLoadBalancer,
+              urlMapRequest: urlMapRequest,
+              backendServiceRequest: backendServiceRequest,
+              httpHealthChecksRequest: httpHealthChecksRequest,
+              groupHealthRequest: groupHealthRequest,
+          )
+
+          compute.targetHttpProxies().get(project, targetProxyName).queue(targetProxyRequest, targetProxyCallback)
+        }
+      }
+    }
+  }
+
+  class TargetProxyCallback<TargetHttpProxy> extends JsonBatchCallback<TargetHttpProxy> implements FailureLogger {
+    GoogleHttpLoadBalancer googleLoadBalancer
+    BatchRequest urlMapRequest
+
+    // Pass through objects
+    BatchRequest backendServiceRequest
+    BatchRequest httpHealthChecksRequest
+    BatchRequest groupHealthRequest
+
+    @Override
+    void onSuccess(TargetHttpProxy targetHttpProxy, HttpHeaders responseHeaders) throws IOException {
+      def urlMapURL = targetHttpProxy?.urlMap
+      if (urlMapURL) {
+        def urlMapName = Utils.getLocalName(urlMapURL)
+        def urlMapCallback = new UrlMapCallback(
+            googleLoadBalancer: googleLoadBalancer,
+            backendServiceRequest: backendServiceRequest,
+            httpHealthChecksRequest: httpHealthChecksRequest,
+            groupHealthRequest: groupHealthRequest,
+        )
+        compute.urlMaps().get(project, urlMapName).queue(urlMapRequest, urlMapCallback)
+      }
+    }
+  }
+
+  class UrlMapCallback<UrlMap> extends JsonBatchCallback<UrlMap> implements FailureLogger {
+    GoogleHttpLoadBalancer googleLoadBalancer
+    BatchRequest backendServiceRequest
+
+    // Pass through objects
+    BatchRequest httpHealthChecksRequest
+    BatchRequest groupHealthRequest
+
+    @Override
+    void onSuccess(UrlMap urlMap, HttpHeaders responseHeaders) throws IOException {
+      // TODO(jacobkiefer): De-dupe backend services?
+      // Default service is mandatory.
+      def backendServiceCallback = new BackendServiceCallback(
+          googleLoadBalancer: googleLoadBalancer,
+          httpHealthChecksRequest: httpHealthChecksRequest,
+          groupHealthRequest: groupHealthRequest,
+      )
+      compute.backendServices()
+          .get(project, Utils.getLocalName(urlMap.defaultService))
+          .queue(backendServiceRequest, backendServiceCallback)
+
+      urlMap.pathMatchers?.each { PathMatcher pathMatcher ->
+        urlMap.hostRules?.each { HostRule hostRule ->
+          if (hostRule.pathMatcher && hostRule.pathMatcher == pathMatcher.name) {
+            def gPathMatcher = new GooglePathMatcher(
+                defaultService: Utils.getLocalName(pathMatcher.defaultService)
```

Comment:
`Utils.getLocalName(pathMatcher.defaultService)` is called several times. Can just capture it in a local variable.